src/components/views/EditUserPreferencePage.js

```diff
@@ -1,13 +1,209 @@
+    const [genderPreference, setGenderPreference] = React.useState({
+        MALE: true,
+        FEMALE: false,
+        OTHER: false,
+    });
+    const { MALE, FEMALE, OTHER } = genderPreference;
+    const noGenderSelectedError = [MALE, FEMALE, OTHER].filter((v) => v).length === 0;
+
+
+    //Gets all the preferences of the user
+    useEffect(() => {
+        async function getPreferences(){
+            try{
+                const response = await api.get('/users/'+id)
+                setGenderPreference({
+                    MALE: response.data.genderPreferences.includes("MALE"),
+                    FEMALE: response.data.genderPreferences.includes("FEMALE"),
+                    OTHER: response.data.genderPreferences.includes("OTHER"),
+                })
+                setAgePreference([response.data.minAge, response.data.maxAge])
+                //TODO: Add distance preferences for the user when GeoLocation API gets installed
+            } catch (error) {
+                console.error("Details:", error);
+                alert("Invalid Input:\n " + handleError(error));
+            }
+        }
+        getPreferences()
+    }, []);
+
+
+    //Method to change the age for ageRangeslider
+    const handleChangeAge = (event, newValue, activeThumb) => {
+        if (!Array.isArray(newValue)) {
+            return;
+        }
+
+        if (activeThumb === 0) {
+            setAgePreference([Math.min(newValue[0], agePreference[1]), agePreference[1]]);
+            console.log(agePreference)
+        } else {
+            setAgePreference([agePreference[0], Math.max(newValue[1], agePreference[0])]);
+            console.log(agePreference)
+        }
+    };
+
+    //Method to changes distances for distance range slider
+    const handleChangeDistance = (event, newValue, activeThumb) => {
+        if (!Array.isArray(newValue)) {
+            return;
+        }
+
+        if (activeThumb === 0) {
+            setDistance([Math.min(newValue[0], distance[1]), distance[1]]);
+            console.log(distance)
+        } else {
+            setDistance([distance[0], Math.max(newValue[1], distance[0])]);
+            console.log(distance)
+        }
+    };
+
+    //Method to handle changes for the checkboxes gender
+    const handleChangeGender = (event) => {
+        setGenderPreference({
+            ...genderPreference,
+            [event.target.name]: event.target.checked,
+        });
+    };
+
+
+    let editPreferences = (
+        <div className="login form">
+            <h2 className="login title"> Set Preferences </h2>
+            <div className="login container-title">Edit Age Preferences:</div>
+            <Slider
+                value={agePreference}
+                onChange={handleChangeAge}
+                valueLabelDisplay="auto"
+                min={18}
+                max={99}
+                disableSwap
+            />
+                Mininum Age: {agePreference[0]} - Maximum Age: {agePreference[1]}
```

Comment:
It's a bit much text here. Would it be sufficient to have `{agePreference[0]} - {agePreference[1]}` ?